File: 剑指offer/easy/20210914/majorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candiate,count=0;
        for(int i=0;i<nums.size();i++){
            if(count==0){
                candiate=nums[i];
            }
            if(candiate==nums[i]){
                count++;
            }else{
                count--;
            }
        }
        return candiate;
    }
};

File: Leetcode/leetcodehot100/20211025/majorityElement.go
package _0211025

func majorityElement(nums []int) int {
	count, num := 0, 0
	for _, value := range nums {
		if count == 0 {
			num = value
		}
		if value == num {
			count++
		} else {
			count--
		}
	}
	return num
}
